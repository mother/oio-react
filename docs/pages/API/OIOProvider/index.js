import React from 'react'
import APITableBlock from '../../../components/APITableBlock'
import HeaderBlock from '../../../components/HeaderBlock'
import SpecimenBlock from '../../../components/SpecimenBlock'
import OIOProviderSpecimen from '../../../specimens/OIOProviderComponent'
import oioSpec from '../../../../src/OIOProvider/spec'

const OIOProviderPage = () => (
   <div>
      <HeaderBlock title="OIOProvider" />
      <SpecimenBlock specimen={OIOProviderSpecimen} />
      <APITableBlock spec={oioSpec} title="Component API" />
   </div>
)

export default OIOProviderPage
