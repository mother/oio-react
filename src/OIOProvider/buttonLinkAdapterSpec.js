export default [{
   name: 'isActive',
   type: 'Function',
   default: 'See Description',
   description: 'A function that is passed in a link. This function must return a boolean that represents whether the link is currently active. The default function does a simple check based on <code>window.location.pathname</code>. This is only by <code>ListMenuButton</code> and <code>TabMenuButton</code>, but not <code>Button</code>.'
}, {
   name: 'render',
   type: 'Function',
   default: 'See Description',
   description: 'A function that is passed the props passed in to the <code>Button</code> or <code>ListMenuButton</code> element. This function must return a React element. The default function simply returns <code>&lt;a href={linkTo}&gt;{children}&lt;/a&gt;</code>.'
}]
