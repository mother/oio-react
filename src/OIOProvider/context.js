import React from 'react'

const OIOContext = React.createContext()

const withOIOContext = WrappedComponent => React.forwardRef((props, ref) => (
   <OIOContext.Consumer>
      {context => (
         <WrappedComponent {...props} OIOContext={context} ref={ref} />
      )}
   </OIOContext.Consumer>
))

export { OIOContext as default, withOIOContext }
