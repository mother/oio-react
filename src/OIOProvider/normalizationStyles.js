import { css } from '@emotion/core'

const normalizationStyles = css`
   *{
      -webkit-tap-highlight-color: rgba(0,0,0,0);
      -webkit-tap-highlight-color: transparent; // "turn off" link highlight
   }

   *, *:before, *:after {
      -webkit-box-sizing: inherit;
      -moz-box-sizing: inherit;
      box-sizing: inherit;
   }

   html, body, div, span, applet, object, iframe,
   h1, h2, h3, h4, h5, h6, p, blockquote, pre,
   a, abbr, acronym, address, big, cite, code,
   del, dfn, em, img, ins, kbd, q, s, samp,
   small, strike, strong, sub, sup, tt, var,
   b, u, i, center,
   dl, dt, dd, ol, ul, li,
   fieldset, form, label, legend,
   table, caption, tbody, tfoot, thead, tr, th, td,
   article, aside, canvas, details, embed,
   figure, figcaption, footer, header, hgroup,
   menu, nav, output, ruby, section, summary,
   time, mark, audio, video {
      margin: 0;
      padding: 0;
      border: 0;
      vertical-align: baseline;
   }

   html {
     -webkit-box-sizing: border-box;
     -moz-box-sizing: border-box;
     box-sizing: border-box;
   }

   a {
      text-decoration: none;
      outline: none;
      border: none;
      cursor: pointer;
      -webkit-tap-highlight-color:rgba(0,0,0,0);
   }

   a:active {
      outline: none;
   }

   a:focus {
      -moz-outline-style: none;
      outline:0;
   }

   img {
      display: block;
      border: none;
      margin: 0;
      padding: 0;
   }

   table {
      margin: 0;
      padding: 0;
      border-collapse: collapse;
      border: none;
   }

   th, td {
      text-align: left;
      padding: 0;
   }

   td {
      vertical-align: top;
   }

   input, textarea {
      margin: 0;
      padding: 0;
      border: 0;
      outline: 0;
      border-radius: 0;
      -webkit-appearance: none;
   }
`

export default normalizationStyles
