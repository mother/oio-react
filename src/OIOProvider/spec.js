export default [{
   name: 'buttonLinkAdapter',
   type: 'Object',
   default: '-',
   description: 'See below'
}, {
   name: 'className',
   type: 'String',
   default: '-',
   description: 'Pass a <code>className</code> to customize className of the OIOProvider'
}, {
   name: 'fontFamily',
   type: 'String',
   default: '<code>sans-serif</code>',
   description: 'Default font-family used for all text inside this OIOProvider'
}, {
   name: 'fontSize',
   type: 'String',
   default: '<code>16px</code>',
   description: 'Default font-size used inside this OIOProvider component'
}, {
   name: 'id',
   type: 'String',
   default: '<code>oio-container-{timestamp}</code>',
   description: 'Container DOM Element ID'
}, {
   name: 'style',
   type: 'Object',
   default: '<code>{}</code>',
   description: 'Pass a <code>style</code> object to customize inline style of the OIOProvider'
}]
