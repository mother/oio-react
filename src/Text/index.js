import React, { Component } from 'react'
import { css, cx } from 'emotion'
import PropTypes from 'prop-types'
import generateStyleObject from '../utils/generators'

@generateStyleObject(({
   autoScale,
   baseFontSize,
   baseAutoScaleFontSize,
   color,
   fontFamily,
   scaleRatio,
   size,
   uppercase,
   weight
}) => {
   // Dynamically calculate font size
   const floatScaleRatio = parseFloat(scaleRatio)
   const adjustedTextSize = size < 1 ? floatScaleRatio * size : floatScaleRatio ** size
   let textSizeCoefficient = autoScale ? baseAutoScaleFontSize : baseFontSize
   textSizeCoefficient = uppercase ? textSizeCoefficient *= 0.9 : textSizeCoefficient

   return ({
      color,
      fontFamily,
      fontSize: `${textSizeCoefficient * adjustedTextSize}${autoScale ? 'vw' : 'px'}`,
      fontWeight: weight,
      lineHeight: size > 5 ? '120%' : '130%',
      textTransform: uppercase && 'uppercase'
   })
})

export default class Text extends Component {
   /* eslint-disable */
   static propTypes = {
      autoScale: PropTypes.bool,
      baseFontSize: PropTypes.string,
      baseAutoScaleFontSize: PropTypes.string,
      calculatedStyleProps: PropTypes.object.isRequired,
      children: PropTypes.node,
      className: PropTypes.string,
      color: PropTypes.string,
      fontFamily: PropTypes.string,
      scaleRatio: PropTypes.string,
      size: PropTypes.string,
      style: PropTypes.object,
      uppercase: PropTypes.bool,
      weight: PropTypes.string
   }
   /* eslint-enable */

   static defaultProps = {
      autoScale: false,
      baseFontSize: '16',
      baseAutoScaleFontSize: '4[a] 2.5[b] 1.75[c] 1.25[d] 1[e] 0.9[f]',
      scaleRatio: '1.125',
      size: '1',
      style: {},
      uppercase: false,
      weight: 'normal'
   }

   render() {
      const styles = {
         ...this.props.style,
         ...this.props.calculatedStyleProps
      }

      return (
         <div className={cx(css(styles), this.props.className)}>
            {this.props.children}
         </div>
      )
   }
}
