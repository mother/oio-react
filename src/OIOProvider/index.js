// =======================================================
// OIOProvider
// All OIO components must be nested inside an OIOProvider
// You can nest OIOProvider inside other OIOProvider
// This component will be a Context provider in future PR
// =======================================================

import React from 'react'
/** @jsx jsx */
import { Global, jsx } from '@emotion/core'
import PropTypes from 'prop-types'
import OIOContext from './context'
import normalizationStyles from './normalizationStyles'

export default class OIOProvider extends React.Component {
   static propTypes = {
      children: PropTypes.node,
      className: PropTypes.string,
      fontFamily: PropTypes.string,
      fontSize: PropTypes.string,
      fontWeightLight: PropTypes.string,
      fontWeightNormal: PropTypes.string,
      fontWeightMedium: PropTypes.string,
      fontWeightSemibold: PropTypes.string,
      fontWeightBold: PropTypes.string,
      formElementAppearance: PropTypes.oneOf(['outline', 'plain']),
      formElementBorderColor: PropTypes.string,
      formElementBorderRadius: PropTypes.string,
      formElementBorderWidth: PropTypes.string,
      formElementBackgroundColor: PropTypes.string,
      formElementFocusBackgroundColor: PropTypes.string,
      formElementFocusBorderColor: PropTypes.string,
      formElementTextColor: PropTypes.string,
      formLabelTextColor: PropTypes.string,
      formLabelTextSize: PropTypes.string,
      formLabelTextTransform: PropTypes.string,
      formLabelTextWeight: PropTypes.string,
      highlightColor: PropTypes.string,
      style: PropTypes.object,
      textSizeScaleRatio: PropTypes.number,
      textSizeMultiplier: PropTypes.number
   }

   static defaultProps = {
      children: null,
      className: undefined,
      fontFamily: 'sans-serif',
      fontSize: '16px',
      fontWeightLight: '300',
      fontWeightNormal: '400',
      fontWeightMedium: '500',
      fontWeightSemibold: '600',
      fontWeightBold: '700',
      formElementAppearance: 'outline',
      formElementBorderColor: '#ddd',
      formElementBorderRadius: '3px',
      formElementBorderWidth: '2px',
      formElementBackgroundColor: 'transparent',
      formElementFocusBackgroundColor: 'rgba(0,0,0,0.01)',
      formElementFocusBorderColor: '#888',
      formElementTextColor: '#333',
      formLabelTextColor: '#444',
      formLabelTextSize: '0.8',
      formLabelTextTransform: 'uppercase',
      formLabelTextWeight: 'bold',
      highlightColor: '#2a383e',
      style: {},
      textSizeScaleRatio: 1.125,
      textSizeMultiplier: 1
   }

   render() {
      const {
         children,
         className,
         fontFamily,
         fontSize,
         fontWeightLight,
         fontWeightNormal,
         fontWeightMedium,
         fontWeightSemibold,
         fontWeightBold,
         formElementAppearance,
         formElementBorderColor,
         formElementBorderRadius,
         formElementBorderWidth,
         formElementBackgroundColor,
         formElementFocusBackgroundColor,
         formElementFocusBorderColor,
         formElementTextColor,
         formLabelTextColor,
         formLabelTextSize,
         formLabelTextTransform,
         formLabelTextWeight,
         highlightColor,
         style,
         textSizeMultiplier,
         textSizeScaleRatio
      } = this.props

      // Props available to OIO Context Consumer
      const contextProps = {
         fontFamily,
         fontSize,
         fontWeightLight,
         fontWeightNormal,
         fontWeightMedium,
         fontWeightSemibold,
         fontWeightBold,
         formElementAppearance,
         formElementBorderColor,
         formElementBorderRadius,
         formElementBorderWidth,
         formElementBackgroundColor,
         formElementFocusBackgroundColor,
         formElementFocusBorderColor,
         formElementTextColor,
         formLabelTextColor,
         formLabelTextSize,
         formLabelTextTransform,
         formLabelTextWeight,
         highlightColor,
         textSizeMultiplier,
         textSizeScaleRatio
      }

      return (
         <OIOContext.Provider value={contextProps}>
            <Global styles={normalizationStyles} />
            <div
               className={className}
               css={{ fontFamily, fontSize, ...style }}>
               {children}
            </div>
         </OIOContext.Provider>
      )
   }
}
