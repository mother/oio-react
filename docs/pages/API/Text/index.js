import React, { Component } from 'react'
import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter'
import { duotoneEarth } from 'react-syntax-highlighter/dist/styles/prism'
import { Text } from '../../../../src'
import TextAutoScaleSpecimen from '../../../specimens/TextAutoScale'
import TextColorsSpecimen from '../../../specimens/TextColors'
import TextSizesSpecimen from '../../../specimens/TextSizes'
import TextTransformsSpecimen from '../../../specimens/TextTransforms'
import TextWeightsSpecimen from '../../../specimens/TextWeights'
import style from '../style.less'

const apiOptions = [{
   name: 'autoScale',
   type: 'Boolean',
   default: '<code>false</code>',
   description: 'When set to <code>true</code>, text will scale dynamically proportionally to the browser screen size',
   responsive: true
}, {
   name: 'className',
   type: 'String',
   default: '-',
   description: 'Pass a <code>className</code> to customize className of the component',
   responsive: false
}, {
   name: 'color',
   type: 'String',
   default: '-',
   description: 'Pass any css compatible color value',
   responsive: true
}, {
   name: 'fontFamily',
   type: 'String',
   default: '-',
   description: 'Apply a specific font-family to the text within this component',
   responsive: true
}, {
   name: 'letterSpacing',
   type: 'String',
   default: '<code>0</code>',
   description: 'Set letter-spacing for Text. Any compatible css letter-spacing values are valid',
   responsive: true
}, {
   name: 'lineHeight',
   type: 'String',
   default: '<code>120%</code>',
   description: 'Set line-height for Text. Any compatible css line-height values are valid',
   responsive: true
}, {
   name: 'size',
   type: 'String',
   default: '<code>3</code>',
   description: 'Set the <code>Text</code> component size using any number. While you may use decimal numbers, it is recommended that whole numbers are used for visual consistency.',
   responsive: true
}, {
   name: 'style',
   type: 'Object',
   default: '<code>{}</code>',
   description: 'Pass a style <code>object</code> to customize inline style of the component',
   responsive: false
}, {
   name: 'transform',
   type: 'String',
   default: '<code>normal</code>',
   description: 'Set text transformations for Text. Any compatible css <code>text-transform</code> values are valid, including: <code>capitalize</code>, <code>lowercase</code>, <code>uppercase</code>',
   responsive: true
}, {
   name: 'weight',
   type: 'String',
   default: '<code>normal</code>',
   description: 'Valid values include: <code>light</code>, <code>normal</code>, <code>medium</code>, <code>semibold</code>, <code>bold</code>',
   responsive: true
}]

export default class TextPage extends Component {
   render() {
      return (
         <div className={style.docs}>
            <div className={style.block}>
               <div className={style.columnHalfText}>
                  <Text size="14" weight="medium">
                     Text
                  </Text>
                  <br />
                  <Text size="5" weight="medium">
                     Text Sizes
                  </Text>
                  <br />
                  <Text>
                     {/* eslint-disable max-len */}
                     You can set the size of the <code>Text</code> component by using the prop <code>size</code>. The default size for the Text component is <code>3</code>. As a general guideline, when using multiple text sizes, it recommended to use sizes at least 3 size units apart (ie. 3,6,9). This helps create better legebility and visual hierarchy.
                     {/* eslint-enable max-len */}
                  </Text>
                  <br />
                  <br />
                  <div className={style.exampleContainer}>
                     <TextSizesSpecimen />
                  </div>
               </div>
               <div className={style.columnHalfCode}>
                  <SyntaxHighlighter
                     language="jsx"
                     style={duotoneEarth}
                     customStyle={{ background: 'transparent' }}>
                     {TextSizesSpecimen.codeSnippet}
                  </SyntaxHighlighter>
               </div>
            </div>
            <div className={style.block}>
               <div className={style.columnHalfText}>
                  <Text size="5" weight="medium">
                     Text Sizes with Autoscale
                  </Text>
                  <br />
                  <Text>
                     {/* eslint-disable max-len */}
                     Sometimes, you might want to have text that scales proportionally to the current browser window size. This is especially useful for larger text elements like titles and headings. <br /><br /> You can make the <code>Text</code> component scale proportionally by setting the prop <code>autoScale</code> to <code>true</code>. Resize your browser window and you can see how the text below scales accordingly.
                     {/* eslint-enable max-len */}
                  </Text>
                  <br />
                  <br />
                  <div className={style.exampleContainer}>
                     <TextAutoScaleSpecimen />
                  </div>
               </div>
               <div className={style.columnHalfCode}>
                  <SyntaxHighlighter
                     language="jsx"
                     style={duotoneEarth}
                     customStyle={{ background: 'transparent' }}>
                     {TextAutoScaleSpecimen.codeSnippet}
                  </SyntaxHighlighter>
               </div>
            </div>
            <div className={style.block}>
               <div className={style.columnHalfText}>
                  <Text size="5" weight="medium">
                     Text Weights
                  </Text>
                  <br />
                  <Text>
                     {/* eslint-disable max-len */}
                     You can set the weight of the <code>Text</code> component by using the prop <code>weight</code>. The default weight for the Text component is <code>normal</code>.
                     {/* eslint-enable max-len */}
                  </Text>
                  <br />
                  <br />
                  <div className={style.exampleContainer}>
                     <TextWeightsSpecimen />
                  </div>
               </div>
               <div className={style.columnHalfCode}>
                  <SyntaxHighlighter
                     language="jsx"
                     style={duotoneEarth}
                     customStyle={{ background: 'transparent' }}>
                     {TextWeightsSpecimen.codeSnippet}
                  </SyntaxHighlighter>
               </div>
            </div>
            <div className={style.block}>
               <div className={style.columnHalfText}>
                  <Text size="5" weight="medium">
                     Text Color
                  </Text>
                  <br />
                  <Text>
                     {/* eslint-disable max-len */}
                     You can set the color of the <code>Text</code> component by using the prop <code>color</code>. Any css compatible color values are valid.
                     {/* eslint-enable max-len */}
                  </Text>
                  <br />
                  <br />
                  <div className={style.exampleContainer}>
                     <TextColorsSpecimen />
                  </div>
               </div>
               <div className={style.columnHalfCode}>
                  <SyntaxHighlighter
                     language="jsx"
                     style={duotoneEarth}
                     customStyle={{ background: 'transparent' }}>
                     {TextColorsSpecimen.codeSnippet}
                  </SyntaxHighlighter>
               </div>
            </div>
            <div className={style.block}>
               <div className={style.columnHalfText}>
                  <Text size="5" weight="medium">
                     Text Transforms
                  </Text>
                  <br />
                  <Text>
                     {/* eslint-disable max-len */}
                     You can set text transformations on <code>Text</code> content to be capitalized or uppercase.
                     {/* eslint-enable max-len */}
                  </Text>
                  <br />
                  <br />
                  <div className={style.exampleContainer}>
                     <TextTransformsSpecimen />
                  </div>
               </div>
               <div className={style.columnHalfCode}>
                  <SyntaxHighlighter
                     language="jsx"
                     style={duotoneEarth}
                     customStyle={{ background: 'transparent' }}>
                     {TextTransformsSpecimen.codeSnippet}
                  </SyntaxHighlighter>
               </div>
            </div>
            <div className={style.block}>
               <div className={style.columnFull}>
                  <Text size="5" weight="medium">
                     Component API
                  </Text>
                  <br />
                  <br />
                  <table className={style.table}>
                     <thead>
                        <tr>
                           <th width="150px">Prop</th>
                           <th width="150px">Prop Type</th>
                           <th width="150px">Default Value</th>
                           <th>Description/Options</th>
                           <th>Responsive</th>
                        </tr>
                     </thead>
                     <tbody>
                        {apiOptions.map(option => (
                           <tr key={option.name}>
                              <td><b>{option.name}</b></td>
                              <td>{option.type}</td>
                              <td>
                                 <div dangerouslySetInnerHTML={{ __html: option.default }} />
                              </td>
                              <td>
                                 <div dangerouslySetInnerHTML={{ __html: option.description }} />
                              </td>
                              <td>
                                 {option.responsive ? <div>Yes</div> : <div>No</div>}
                              </td>
                           </tr>
                        ))}
                     </tbody>
                  </table>
               </div>
            </div>
         </div>
      )
   }
}
