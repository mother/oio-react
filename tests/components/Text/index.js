import React from 'react'
import { render } from 'react-testing-library'
import OIOProvider from '../../../src/OIOProvider'
import Text from '../../../src/Text'

test('Text will render at different sizes correctly', () => {
   const { container } = render(
      <OIOProvider>
         <Text size="1[a] 2[b] 3[c] 4[d] 5[e] 6[f]">
            Responsive Sizes
         </Text>
      </OIOProvider>
   )

   expect(container).toMatchSnapshot()
})

test('Text will render at autoScale sizes correctly', () => {
   const { container } = render(
      <OIOProvider>
         <Text size="1[a] 2[b] 3[c] 4[d] 5[e] 6[f]" autoScale>
            AutoScaled Responsive Sizes
         </Text>
      </OIOProvider>
   )

   expect(container).toMatchSnapshot()
})

test('Text will render at different font weights correctly', () => {
   const { container } = render(
      <OIOProvider>
         <Text weight="bold[a] semibold[b] medium[c] normal[d] light[e-f]">
            Text Weights
         </Text>
      </OIOProvider>
   )

   expect(container).toMatchSnapshot()
})

test('Text will render color correctly', () => {
   const { container } = render(
      <OIOProvider>
         <Text color="red[a-b] #ff0000[c-d] rgba(255,0,0,1)[e-f]">Red Text</Text>
      </OIOProvider>
   )

   expect(container).toMatchSnapshot()
})

test('Text will render text transforms correctly', () => {
   const { container } = render(
      <OIOProvider>
         <Text uppercase>Uppercase Text</Text>
         <Text capitalize>capitalized text</Text>
         <Text>Standard Case Text</Text>
      </OIOProvider>
   )

   expect(container).toMatchSnapshot()
})

test('Invalid prop values passed to Text component will be ignored', () => {
   const { container } = render(
      <OIOProvider>
         <Text
            size="The Wrong Size"
            weight="fakeWeight">
            Non Existing Font Weight
         </Text>
      </OIOProvider>
   )

   expect(container).toMatchSnapshot()
})
