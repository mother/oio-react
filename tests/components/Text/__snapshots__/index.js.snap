// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid prop values passed to Text component will be ignored 1`] = `
.emotion-1 {
  font-family: sans-serif;
  font-size: 16px;
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-0 {
    line-height: 130%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-0 {
    line-height: 130%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-0 {
    line-height: 130%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-0 {
    line-height: 130%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-0 {
    line-height: 130%;
  }
}

@media (min-width:1700px) {
  .emotion-0 {
    line-height: 130%;
  }
}

<div>
  <div
    class="emotion-1"
    id="oio"
  >
    <div
      class=" emotion-0"
    >
      Non Existing Font Weight
    </div>
  </div>
</div>
`;

exports[`Text will render at autoScale sizes correctly 1`] = `
.emotion-1 {
  font-family: sans-serif;
  font-size: 16px;
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-0 {
    font-size: 2.8125vw;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-0 {
    font-size: 2.278125vw;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-0 {
    font-size: 2.1357421875vw;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-0 {
    font-size: 1.601806640625vw;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-0 {
    font-size: 1.3515243530273438vw;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:1700px) {
  .emotion-0 {
    font-size: 1.2670540809631348vw;
    font-weight: 400;
    line-height: 110%;
  }
}

<div>
  <div
    class="emotion-1"
    id="oio"
  >
    <div
      class=" emotion-0"
    >
      AutoScaled Responsive Sizes
    </div>
  </div>
</div>
`;

exports[`Text will render at different font weights correctly 1`] = `
.emotion-1 {
  font-family: sans-serif;
  font-size: 16px;
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 700;
    line-height: 130%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 600;
    line-height: 130%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 500;
    line-height: 130%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 300;
    line-height: 130%;
  }
}

@media (min-width:1700px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 300;
    line-height: 130%;
  }
}

<div>
  <div
    class="emotion-1"
    id="oio"
  >
    <div
      class=" emotion-0"
    >
      Text Weights
    </div>
  </div>
</div>
`;

exports[`Text will render at different sizes correctly 1`] = `
.emotion-1 {
  font-family: sans-serif;
  font-size: 16px;
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-0 {
    font-size: 12.375px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-0 {
    font-size: 13.921875px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-0 {
    font-size: 17.619873046875px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-0 {
    font-size: 19.822357177734375px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:1700px) {
  .emotion-0 {
    font-size: 22.300151824951172px;
    font-weight: 400;
    line-height: 110%;
  }
}

<div>
  <div
    class="emotion-1"
    id="oio"
  >
    <div
      class=" emotion-0"
    >
      Responsive Sizes
    </div>
  </div>
</div>
`;

exports[`Text will render color correctly 1`] = `
.emotion-1 {
  font-family: sans-serif;
  font-size: 16px;
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-0 {
    color: red;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-0 {
    color: red;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-0 {
    color: #ff0000;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-0 {
    color: #ff0000;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-0 {
    color: rgba(255,0,0,1);
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:1700px) {
  .emotion-0 {
    color: rgba(255,0,0,1);
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
  }
}

<div>
  <div
    class="emotion-1"
    id="oio"
  >
    <div
      class=" emotion-0"
    >
      Red Text
    </div>
  </div>
</div>
`;

exports[`Text will render text transforms correctly 1`] = `
.emotion-3 {
  font-family: sans-serif;
  font-size: 16px;
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
    text-transform: uppercase;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
    text-transform: uppercase;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
    text-transform: uppercase;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
    text-transform: uppercase;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
    text-transform: uppercase;
  }
}

@media (min-width:1700px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
    text-transform: uppercase;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
    text-transform: capitalize;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
    text-transform: capitalize;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
    text-transform: capitalize;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
    text-transform: capitalize;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
    text-transform: capitalize;
  }
}

@media (min-width:1700px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
    text-transform: capitalize;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
  }
}

@media (min-width:1700px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 130%;
  }
}

<div>
  <div
    class="emotion-3"
    id="oio"
  >
    <div
      class=" emotion-0"
    >
      Uppercase Text
    </div>
    <div
      class=" emotion-1"
    >
      capitalized text
    </div>
    <div
      class=" emotion-2"
    >
      Standard Case Text
    </div>
  </div>
</div>
`;
